LEMS/LEMS_EnergyCalcs.py

```diff
@@ -724,6 +724,157 @@ def LEMS_EnergyCalcs(inputpath,outputpath,logpath):
         elif uval['eff_w_char_weighted'].n >= 50:
             uval[name] = 'Tier 5'
 
+    ##########################################################################
+    #Calculations for full period
+    name = 'wood_dry_mass'
+    names.append(name)
+    units[name] = 'kg'
+    uval[name] = []
+    NCV = []
+    ECV = []
+
+    name = 'wood_mass'
+    names.append(name)
+    units[name] = 'kg'
+    uval[name] = 0
+
+    name = 'char_mass_initial'
+    names.append(name)
+    units[name] = 'lb'
+    uval[name] = 0
+
+    name = 'char_mass_final'
+    names.append(name)
+    units[name] = 'lb'
+    uval[name] = 0
+
+    for n, fuel in enumerate(fuels):
+        identifier = f"_{str(n+1)}"
+        if uval[f"fuel_Cfrac{identifier}"] < 0.75: #If fuel is wood
+            for phase in phases:
+                massname = f"fuel_dry_mass_{phase}{identifier}" #fuel_mass_phase_#
+                if type(uval[massname]) is not str:
+                    uval['wood_dry_mass'].append(uval[massname]) #Sum all wood masses for total wood input
+                massname = f"fuel_mass_{phase}{identifier}"  # fuel_mass_phase_#
+                if type(uval[massname]) is not str:
+                    uval['wood_mass'] = uval['wood_mass'] + uval[massname]
+            NCV.append(uval[f'fuel_net_calorific_value{identifier}'])
+            ECV.append(uval[f'fuel_effective_calorific_value{identifier}'])
+
+        elif uval[f"fuel_Cfrac{identifier}"].n > 0.75: #If fuel is charcoal
+            for phase in phases:
+                imassname = f"initial_fuel_mass{identifier}_{phase}"
+                fmassname = f"final_fuel_mass{identifier}_{phase}"
+                if uval['char_mass_initial'] == 0: #only record the first charcoal input
+                    uval['char_mass_initial'] = uval[imassname]
+                if uval[fmassname] != 0: #only record the last charcoal output
+                    uval['char_mass_final'] = uval[fmassname]
+            char_NCV = uval[f'fuel_net_calorific_value{identifier}']
+            char_ECV = uval[f'fuel_effective_calorific_value{identifier}']
+
+    name = 'char_mass_lb'
+    names.append(name)
+    units[name] = 'lb'
+    uval[name] = uval['char_mass_final'] - uval['char_mass_initial']

```

Comment:
When processing an IDC test with only a high power phase, had to put the high power charcoal start value in L1 charcoal start and the high power charcoal end value in L5 charcoal end.

Reply:
@sameb112 Code has been updated to be more dynamic, see lines 766 to 775. Char mass initial (full period) is now detected by finding the first phase where there's a difference in char mass and taking the initial mass value. Char mass final (full period) finds the last phase where there's a difference in char mass and takes the final mass value